modules/view_establishment/predicates.py

```diff
@@ -0,0 +1,344 @@
+        """Perform the action corresponding to the current situation."""
+        # Phase 0, waiting for a next view
+        if (phase == 0):
+            return self.automation_phase_0(type, case)
+
+        # Phase 1, in a view change
+        elif(phase == 1):
+            return self.automation_phase_1(type, case)
+
+        # Not a valid phase
+        else:
+            raise Exception('Not a valid phase: {}'.format(phase))
+
+    def automation_phase_0(self, type, case):
+        """Perform the action corresponding to the current case of phase 0."""
+        # Predicates
+        if(type == "pred"):
+            # True if a view pair is adoptable but is not the view of
+            # processor i
+            if(case == 0):
+                for processor_id, view_pair in enumerate(self.views):
+                    if (self.transit_adopble(processor_id, 0, "Follow") and
+                        self.views[self.id].get("current") !=
+                            view_pair.get("current")):
+                        self.view_pair_to_adopt = view_pair
+                        return True
+                return False
+
+            # True if a view change was instructed by Primary Monitoring
+            elif(case == 1):
+                return (self.vChange and self.establishable(0, "Follow"))
+
+            # Monitoring/Waiting for more processors acknowledgement
+            elif(case == 2):
+                return (self.transit_adopble(self.id, 0, "Remain") or
+                        self.views[self.id] == self.RST_PAIR)
+
+            # True if there is no adoptable view (predicates for other cases
+            #  are false)
+            elif(case == 3):
+                return True
+
+            # Not a valid case
+            else:
+                raise Exception('Not a valid case: {}'.format(case))
+
+        # Actions
+        elif(type == "act"):
+
+            # Adopt the new view
+            if(case == 0):
+                if(self.view_pair_to_adopt):
+                    self.adopt(self.view_pair_to_adopt)
+                    self.view_pair_to_adopt = None
+                    self.view_module.next_phs()
+                    self.reset_v_change()
+                    return "Adopted new view"
+                else:
+                    raise Exception("Not a valid view pair to adopt")
+                # TODO What do return? and to we need a if statement
+                # for view_pair_to_adopt?
+
+            # Increment view (next view)
+            elif(case == 1):
+                self.next_view()
+                self.view_module.next_phs()
+                return "Incremented view"
+                # TODO What do return?
+
+            # No action and reset the v_change-variable
+            elif(case == 2):
+                self.reset_v_change()
+                return "No action"
+
+            # Full reset
+            elif(case == 3):
+                return self.reset_all()
```

Comment:
Should call `reset_v_change` as well, maybe add a TODO here to remember to add call to `reset_v_change` when it is implemented

Reply:
Added the call and updated corresponding test